### ISSUE TITLE
Pod dependency with weak references

### ISSUE THREAD

**Issue by BobDG** [CocoaPods/CocoaPods#6961]
# Report

## What did you do?

I'm trying to push a private pod to a private repo. The pod uses several other pods as a dependency and one of these is 'BlazeMarkdownCell'. The command I used to check my podspec is this:

pod lib lint --private --allow-warnings --sources=[Private Repo Github URL],master --use-libraries --verbose

## What did you expect to happen?

I expected that the podspec would validate just fine.

## What happened instead?

It fails to validate according to this error:
ERROR | [iOS] xcodebuild:  BlazeMarkdownCell/BlazeMarkdownCell/TSMarkdownParser/TSMarkdownParser.m:82:5: error: cannot create __weak reference because the current deployment target does not support weak references

Now I've researched this issue a lot online and could only find that I could possibly set some build settings using the xcconfig setting of the podspec. So I've tried this:
  s.xcconfig = {
	'CLANG_ENABLE_OBJC_WEAK' => 'YES'
	}

In both the podspec of the private pod and both the podspec of the dependency pod that's causing the issue. Unfortunately they seem to have no effect. Is this perhaps an issue for dependency pods that's not resolved yet?

### Stack

```
   CocoaPods : 1.3.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.6 (16G29)
       Xcode : 8.3.3 (8E3004b)
         Git : git version 2.11.0 (Apple Git-81)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 23dda4de6ba34b39a5375d3cfdd4971728593f6c     
               [Private Repo] - [Private Repo URL]
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.2.0
cocoapods-try         : 1.1.0
```

**Comment by dnkoutso** [comment]
Can you use `s.pod_target_xcconfig` instead of `s.xcconfig`?

**Comment by dnkoutso** [comment]
This does not seem a problem with CocoaPods the library itself though. I am going to close this but feel free to comment here.

**Comment by BobDG** [comment]
@dnkoutso I've tried using your suggestion but unfortunately it still doesn't work, I get the same error. Why do you think this is not an issue with Cocoapods the library? I was able to create the CocoaPod of BlazeMarkdownCell without any issues like this but when I use it as a dependency I DO get an error when trying to create a new CocoaPod. Isn't that some kind of linting/building issue within the Cocoapods library? That only happens when you use weak references within pod dependencies in your podspec?

**Comment by dnkoutso** [comment]
Hm @BobDG I may have pulled the trigger a bit fast. Can you upload a sample project? Those make it much easier to debug.

**Comment by BobDG** [comment]
@dnkoutso 
Ah I found the issue. When I created the example project using the BlazeMarkdownCell pod it didn't compile correctly with the same error. Then I found out somehow it thought my pod had a deployment target below iOS6. Actually, it didn't have any deployment target at all. It was in my podspec though but apparently it wasn't pushed correctly. And if you have no deployment target in your podspec it thinks it should work for all possible iOS versions I guess? Anyway, now I've set a specific deployment target in my dependency pod and now it validates without issues. Thanks for looking into this and making me create a sample project. Will do that next time before creating an issue :)
